src/transaction/components/TransactionProvider.tsx

```diff
@@ -68,7 +74,24 @@ export function TransactionProvider({
 
   const { switchChainAsync } = useSwitchChain();
 
-  // Hooks that depend from Core Hooks
+  // Validate `calls` and `contracts` props
+  if (!contracts && !calls) {
+    setErrorMessage('One of contracts or calls must be provided.');
```

Comment:
Everytime we trigger a new error, let's do that by using the LifeCycleStatus.

So that also `onStatus` gets triggered.